index.js

```diff
@@ -233,7 +238,7 @@ NYC.prototype.walkAllFiles = function (dir, visitor) {
 }
 
 NYC.prototype._maybeInstrumentSource = function (code, filename, relFile) {
-  var instrument = this.exclude.shouldInstrument(filename, relFile)
+  var instrument = (!this.instrument && this.all) || this.exclude.shouldInstrument(filename, relFile)
```

Comment:
`But users may be surprised/confused at the need to include their generated files` 👍 

I _think_ the logic is correct if we just call ` this.exclude.shouldInstrument(filename, relFile)` without any changes. Keep in mind that when we run `nyc --all` we iterate over all the files in the project using `glob`:

```js
 glob.sync(pattern, {cwd: dir, nodir: true, ignore: this.exclude.exclude}).forEach(function (filename) {
    visitor(filename)
  })
```

As long as folks setup their `include`/`exlude` rules appropriately, this means that we'll pull in the coverage headers regardless of whether they've been pre-generated and stored to disk, or in the case of `babel-register` are being plopped in on the fly.

So far folks have been having good luck playing with this new feature:

https://github.com/istanbuljs/babel-plugin-istanbul/issues/4

Great work @motiz88